resolve-workgroup/decisions/AYD-RE-002-Reviewable-intermediate-files.md

```diff
@@ -0,0 +1,79 @@
+# AYD-RE-002 Generating intermediate files for reviewable representations
+
+- **Status:** Proposed
+- **Impact:**  Medium
+- **Contributors:**:
+  - YNPUT Team <team@ynput.io>,
+  - Jakub Jezek <jakub@ynput.io>
+  - Tony Dorfmeister  (Bepic) @tweak-wtf
+  - Jiri Sindelar (Dazzle Pictures) @jrsndl
+- **Approved by:** -
+
+## Background
+
+To streamline our editorial workflow, we need to automate the creation of video files (used for reviewable/reference representation) that align with our publishing timeline. Currently, this task is manually handled by artists. Automating it will enhance efficiency.
+
+
+## Options
+
+### Option 1
+
+Suppose we were to mirror a similar process to Nuke's Bake stream video, this would mean leveraging Resolve to generate an intermediate video file seamlessly on-the-fly. This newly created intermediate file would then serve as the input for transcoding options such as OIIO and FFMPEG.
+
+Notably, this process should also incorporate baked color grading aesthetics for a more streamlined look, as well as facilitate the display & view.
+
+Additionally, the inclusion of the display and view space ensures that the transcoded video is not restricted and maintains the highest quality, from color to clarity. This arranged method allows us to harness the full potential of transcoding with Resolve, OIIO, and FFMPEG.
+
+This process would mean following steps would need to be (done with current Resolve API and OTIO:
+1. Creating rendering settings `Project.SetRenderSettings()` with predefined temporary file path
+2. Adding job to render queue `Project.AddRenderJob()` and capture return job id
+3. Starting rendering only for job with given id `Project.StartRendering([job_id], isInteractiveMode=False)`
+4. Getting the file from path and using it as source for reviewable representation which will need to be trimmed by clips' lengths
+5. Deleting the temporary file
+6. using trimmed intermediate files as source for OIIO or FFMPEG transcoding
+
+
+#### Pros
+
+- reducing manual work
+- baking color grading look into reference files
+- rendering via GPU will be faster than CPU
```

Comment:
There's nothing in the Option 1 explanation that screams to me that it now uses GPU over CPU? What of the option makes this suddenly run on GPU?

Reply:
Rendering with Resolve's native workflow is using GPU. CPU would be used in case of Option 2 since it would need to use FFMPEG and OIIO.

Reply:
how about having 2 render outputs in resolve "plate" and "baking". resolve can add multiple output passes. not sure if also via API tho.
we would use resolve GPU accelerated render and still be able to post-process the "baking" prores/dnxhr with OIIO/FFMPEG even on the farm